### PR TITLE
Restore splitter persistence

### LINKED ISSUE #8745: Splitter positions are not restored correctly
## Current behaviour

The splitter positions of FormBrowse are not re/stored correctly. At least the CommitInfo panel is shrinked by 8 pixels if displayed beside the RevisionGrid.

## Expected behaviour

CommitInfo width is restored - all splitter positions of course.

## Steps to reproduce

- position CommitInfo beside RevisionGrid
- restart GE

## Did this work in previous version of GitExtensions

yes, before borders were added

## Environment

- Git Extensions 33.33.33
- Build d2830009cd34a660040c66e49a68e7717cd3b2d4
- Git 2.27.0.windows.1 (recommended: 2.28.0 or later)
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 96dpi (no scaling)

### LINKED ISSUE #9000: Toggle left panel status reset after restart
## Current behaviour
Toggle left panel status reset after restart

## Expected behaviour
Toggle left panel status has to be same after restart

## Steps to reproduce
1. turn off left panel
2. restart Git Extensions

## Did this work in previous version of GitExtensions
Yes, but it was long ago. I don't remember the version

## Environment
- Git Extensions 3.6.0.11442
- Build 549f02363b5470696a27a1f002241f5ae4f5c719 (Dirty)
- Git 2.30.2.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 96dpi (no scaling)

### LINKED ISSUE #9089: Left panel toggle state is not stored/restored in 3.5 RC 1
## Current behaviour

Changing toggle for left panel in main window to disabled is not stored and restored at next start

## Expected behaviour

State of left panel toggle should be restored on next start

## Steps to reproduce

- disable left panel
- restart GitExtensions
-> panel is visible again

Split view layout and Commit info position is stored correct

## Did this work in previous version of GitExtensions

Yes: latest stable release (3.4)

## Environment

- Git Extensions 3.5.0.11509
- Build 009d10c765a90aa00e043e40d5117256b793bc8c (Dirty)
- Git 2.31.1.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4300.0
- DPI 144dpi (150% scaling)

### PR DESCRIPTION
**Pull request by RussKie**
Resolves #8745
Resolves #9000
Resolves #9089

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by ivangrek** [comment]
What changed?

File: UnitTests/GitUI.Tests/SplitterManagerTest.cs [hunk 1]
```diff
@@ -6,216 +6,215 @@
...
+                splitManager.AddSplitter(splitter, splitterName);
+                int splitterNewWidth = splitterWidth + deltaWidth;
+                splitter.Width = splitterNewWidth;
+                splitter.FixedPanel = FixedPanel.Panel2;
+                splitManager.RestoreSplitters();
+
+                // assert splitter moved by the width delta
+                // Note: if the width of the splitter control itself is not regarded,
+                // this splitter control would move to the left every time the splitter container is restored.
+                splitter.SplitterDistance.Should().Be(splitterDistance + deltaWidth + splitter.SplitterWidth - 2);
```
Comment (RussKie) [inline UnitTests/GitUI.Tests/SplitterManagerTest.cs:112]: `-2` here

File: UnitTests/GitUI.Tests/SplitterManagerTest.cs [hunk 2]
```diff
@@ -6,216 +6,215 @@
...
+                // assert
+                if (applyMinSize)
+                {
+                    splitter.SplitterDistance.Should().Be(splitterDistance);
+                }
+                else
+                {
+                    // Note: if the width of the splitter control itself is not regarded,
+                    // this splitter control would move to the left every time the splitter container is restored.
+                    splitter.SplitterDistance.Should().Be(splitterDistance + deltaWidth + splitter.SplitterWidth - 2);
```
Comment (RussKie) [inline UnitTests/GitUI.Tests/SplitterManagerTest.cs:159]: ...and `-2` here

File: UnitTests/GitUI.Tests/SplitterManagerTest.cs (hunk 1 above)
Comment (ivangrek) [reply UnitTests/GitUI.Tests/SplitterManagerTest.cs:112]: I did it but it didn't help.
The situation is that nested elements affect the sizes in a cascade.

**Comment by h0lg** [comment]
Just updated to 3.5.2.12435
- Build 77413699a9c25b160a6a622311529372b5d87f5b
- Git 2.32.0.windows.2

and the issue persists. Closing the left panel on smaller screens is getting kind of old... Is this supposed to be in there?

**Comment by mstv** [comment]
It works well for me.

> is getting kind of old...

What do mean? Screenshots are worth a thousand words.

**Comment by h0lg** [comment]
I meant that I'm growing tired of closing the left sidebar every time I open GitExtensions because having it closed/disabled doesn't seem to persist and on smaller screens I prefer having more space in the browser fro reading. I usually switch between (known) branches using [Ctrl+.] anyway, so I only use sidebar for searching when I don't remember a branch name.

**Comment by mstv** [comment]
You could check the setting `FormBrowse.MainSplitContainer_Panel1Collapsed` in `%AppData%\Roaming\GitExtensions\GitExtensions\GitExtensions.settings`.

How do you close GE? How many instances at a time? The last one wins (whichever it is).

**Comment by h0lg** [comment]
```
  <item>
    <key>
      <string>FormBrowse.MainSplitContainer_Panel1Collapsed</string>
    </key>
    <value>
      <string>true</string>
    </value>
  </item>
```
The value is and remains `true`, whether I close GE with the left side bar open or collapsed. And it invariably starts up opening the side bar again.

I close GE either using [Alt+F4] or via the window's close icon. I have mostly one, sometimes 2 and rarely more instances open. I understand that the last instance closing persists its own settings.

**Comment by mstv** [comment]
Indeed, the Left Panel collapsed status is not correctly taken into account if the dashboard is opened on startup.
Partial workaround: `Open last working directory on startup`. Hit `Ctrl+W` to go to the dashboard.
image

**Comment by h0lg** [comment]
> Partial workaround: `Open last working directory on startup`.

That work-around is completely acceptable for me, thanks!

**Comment by RussKie** [comment]
We currently don't persist the state of splitters/size of panels. https://github.com/gitextensions/gitextensions/pull/9106 is supposed to address this.

**Pull request merged**